### PR TITLE
[CM-586] Consider expires header when caching idres responses

### PR DESCRIPTION
**Pull request by mschuwalow**
https://liveintent.atlassian.net/browse/CM-587

Short description if any.

Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

### CODE REVIEW

**Review by 3link — APPROVED** [review]

**Comment by 3link** [comment]
Rather related to the implementation of the feature in general: There are now two ways of defining what needs to be resolved in LC:
- via the `requestedAttributes` config parameter, which results in `resolve` parameters being send to IdRes: https://github.com/LiveIntent/live-connect/pull/76/files#diff-c6b1818a2e0b7118e720980ab2d8fc3af8270c3e72d85837820f769633a29383R100
- via the `additionalParams` parameter of the `resolve` function which also results in `resolve` parameters being send to IdRes (additionalParams can be something like `{resolve: ['nonId', 'md5', 'uid2']}):` https://github.com/LiveIntent/live-connect/pull/76/files#diff-c6b1818a2e0b7118e720980ab2d8fc3af8270c3e72d85837820f769633a29383R133

Let's get rid of the `requestedAttributes` config parameter. Prebid's user id module could then translate the `requestedAttributesOverrides` config into `additionalParams` for the resolve function.

**Comment by mschuwalow** [comment]
My understanding is that additionalParams is a bit of an escape hatch. We are already exposing other configuration parameters that are more clear / specific for other parameters -- I don't really see a reason to depart from this pattern for this parameter in particular

File: src/idex/common.js [hunk 1]
```diff
@@ -163,3 +165,12 @@ export function makeIdentityResolver (config, calls, cache) {
     }
   }
 }
+
+function responseExpires (response) {
+  if (response && typeof response.getResponseHeader === 'function') {
```
Comment (wi101) [inline src/idex/common.js:170]: You can use `isFunction` from `utils/types.js`

File: src/idex/common.js [hunk 2]
```diff
@@ -108,25 +108,27 @@ export function makeIdentityResolver (config, calls, cache) {
...
           } catch (ex) {
             console.error('Error parsing response', ex)
             fromError('IdentityResolverParser', ex)
           }
         }
-        cache.set(additionalParams, responseObj)
+
+        const expiresAt = responseExpires(response)
+
+        cache.set(additionalParams, responseObj, expiresAt)
```
Comment (wi101) [inline src/idex/common.js:124]: there will be no issue if the expiresAt is undefined?

File: src/idex/common.js [hunk 3]
```diff
@@ -108,25 +108,27 @@ export function makeIdentityResolver (config, calls, cache) {
...
-            responseObj = JSON.parse(response)
+            responseObj = JSON.parse(responseText)
           } catch (ex) {
             console.error('Error parsing response', ex)
             fromError('IdentityResolverParser', ex)
           }
         }
-        cache.set(additionalParams, responseObj)
+
+        const expiresAt = responseExpires(response)
```
Comment (wi101) [inline src/idex/common.js:122]: should it be?
```suggestion
        const expiresAt = responseExpires(responseObj)
```

Comment (mschuwalow) [reply src/idex/common.js:122]: We need to extract it from the headers. ResponseText is only the body

File: src/idex/common.js (hunk 2 above)
Comment (mschuwalow) [reply src/idex/common.js:124]: the cache handles it properly
`expiresAt || expiresInHours(defaultExpirationHours)`

**Review by wi101 — APPROVED** [review]

**Pull request merged**